### PR TITLE
feat: tekton triggers

### PR DESCRIPTION
**Pull request by srodenhuis**
This PR:

- adds Tekton trigger CRDs
- adds the option to use a triggertemplate instead of a pipelinerun to use a trigger to start the build
- adds all required RBAC to use triggers to the team-ns

### CODE REVIEW

**Review by j-zimnowoda — DISMISSED** [review]

**Review by j-zimnowoda — COMMENTED** [review]
Still we need to test it

File: charts/tekton-triggers/templates/interceptors.yaml [hunk 1]
```diff
@@ -0,0 +1,185 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: tekton-triggers-core-interceptors-certs
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/name: core-interceptors
+    app.kubernetes.io/component: interceptors
```
Comment (j-zimnowoda) [inline charts/tekton-triggers/templates/interceptors.yaml:8]: please split this file to corresponding file names
e.g.: ConfigMap to configmap.yaml

File: charts/tekton-triggers/templates/interceptors.yaml [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    triggers.tekton.dev/release: "v0.25.0"
+# The data is populated at install time.
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tekton-triggers-core-interceptors
+  namespace: tekton-pipelines
+  labels:
+    app.kubernetes.io/name: core-interceptors
```
Comment (j-zimnowoda) [inline charts/tekton-triggers/templates/interceptors.yaml:20]: these labels should be referenced from the _helpers.yaml file

File: charts/tekton-triggers/templates/release.yaml [hunk 3]
```diff
@@ -0,0 +1,858 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (j-zimnowoda) [inline charts/tekton-triggers/templates/release.yaml:2]: Please split this file to smaller chunks

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
There is quite some work to be done here:
1. Bootstrap chart with `helm create tekton-triggers` and then modify it with you templates
2. Reiterate over RBAC and required permissions because current one are quite extensive.
3. Enrich the integrations tests `minimal-with-team.yaml` to accommodate build triggers with some example repo that is hosted at GitHub in redkubes org.

File: charts/team-ns/templates/builds/buildpack.yaml [hunk 4]
```diff
@@ -46,6 +46,45 @@ spec:
...
+            fsGroup: 65532
+        workspaces:
+        - name: shared-data
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+              - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 1Gi
```
[the remark this answers is not in the document]
Comment (srodenhuis) [reply charts/team-ns/templates/builds/buildpack.yaml:79]: the shared data in the workspace is only used for the git clone, not for building images. When git is cloned, the build task uses this for building. The build is done in a container and then pushed. The size is not related to the size of the image that has been build.

File: charts/team-ns/templates/builds/buildpack.yaml [hunk 5]
```diff
@@ -46,6 +46,45 @@ spec:
...
+    - apiVersion: tekton.dev/v1beta1
+      kind: PipelineRun
+      metadata:
+        generateName: buildpacks-build-{{ .name }}-{{ regexReplaceAll "[|._]{1}" .tag "-" }}-
+        labels:
+          tekton.dev/pipeline: buildpacks-build-{{ .name }}
+        annotations:
+          sidecar.istio.io/inject: "false"
+      spec:
+        serviceAccountName: sa-team-{{ $v.teamId }}
```
[the remark this answers is not in the document]
Comment (srodenhuis) [reply charts/team-ns/templates/builds/buildpack.yaml:65]: No. the sa here is the same as in the pipelinerun without a trigger. The `sa-team-{{ $v.teamId }}` has the required secrets needed to push images to harbor

File: charts/team-ns/templates/builds/docker.yaml [hunk 6]
```diff
@@ -44,6 +44,45 @@ spec:
...
+            fsGroup: 65532
+        workspaces:
+        - name: shared-data
+          volumeClaimTemplate:
+            spec:
+              accessModes:
+              - ReadWriteOnce
+              resources:
+                requests:
+                  storage: 1Gi
```
[the remark this answers is not in the document]
Comment (srodenhuis) [reply charts/team-ns/templates/builds/docker.yaml:77]: See previous

File: charts/team-ns/templates/builds/buildpack.yaml [hunk 7]
```diff
@@ -74,6 +113,34 @@ spec:
   - name: docker-credentials
     secret:
       secretName: harbor-pushsecret-builds
+---
+{{- end }}
+{{- if .trigger }}
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: gitea-triggerbinding-{{ .name }}
```
[the remark this answers is not in the document]
Comment (srodenhuis) [reply charts/team-ns/templates/builds/buildpack.yaml:122]: No. All build names are unique

File: charts/team-ns/templates/rbac.yaml [hunk 8]
```diff
@@ -120,3 +120,107 @@ stringData:
...
+  name: tekton-triggers-team-{{ $v.teamId }}
+  namespace: team-{{ $v.teamId }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tekton-triggers-aggregate-view
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
```
[the remark this answers is not in the document]
Comment (srodenhuis) [reply charts/team-ns/templates/rbac.yaml:170]: not so extensive. only for specific resources

File: charts/team-ns/templates/builds/buildpack.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply charts/team-ns/templates/builds/buildpack.yaml:79]: Ok, we found out that layers are stored in host (emptyDir)

File: charts/team-ns/templates/builds/buildpack.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply charts/team-ns/templates/builds/buildpack.yaml:122]: ok, it is responsibility of upper layer to ensure name uniqueness

File: charts/team-ns/templates/builds/buildpack.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply charts/team-ns/templates/builds/buildpack.yaml:65]: thanks

**Review by j-zimnowoda — APPROVED** [review]

**Review by j-zimnowoda — APPROVED** [review]

**Pull request merged**